### DISCUSSION TITLE
Table DateOnly Filter support

### DISCUSSION THREAD

**Discussion by davhev8619** [ant-design-blazor/ant-design-blazor#3778] [Ideas]
I would like to have filter support on DateOnly properties on a Table. I currently have a "date" column in my database. If I use a DateOnly property and try to filter I get NotSupportedException (see your code below). If I use a DateTime property I get this SQL query and it doesn't work either.

DateTime C# - date column in SQL => 
DATEADD(millisecond, CAST(CAST(0 - DATEPART(millisecond, [MyDateColumn]) AS float) AS int), [MyDateColumn]) > '2024-04-04'
=> The datepart millisecond is not supported by date function datepart for data type date.

Your supported filter types:

        return underlyingType switch
        {
            _ when underlyingType.IsEnum              => new EnumFieldFilterType<T>(),
            _ when underlyingType == typeof(byte)     => new NumberFieldFilterType<byte>(),
            _ when underlyingType == typeof(decimal)  => new NumberFieldFilterType<decimal>(),
            _ when underlyingType == typeof(double)   => new NumberFieldFilterType<double>(),
            _ when underlyingType == typeof(short)    => new NumberFieldFilterType<short>(),
            _ when underlyingType == typeof(int)      => new NumberFieldFilterType<int>(),
            _ when underlyingType == typeof(long)     => new NumberFieldFilterType<long>(),
            _ when underlyingType == typeof(sbyte)    => new NumberFieldFilterType<sbyte>(),
            _ when underlyingType == typeof(float)    => new NumberFieldFilterType<float>(),
            _ when underlyingType == typeof(ushort)   => new NumberFieldFilterType<ushort>(),
            _ when underlyingType == typeof(uint)     => new NumberFieldFilterType<uint>(),
            _ when underlyingType == typeof(ulong)    => new NumberFieldFilterType<ulong>(),
            _ when underlyingType == typeof(DateTime) => new DateTimeFieldFilterType(),
            _ when underlyingType == typeof(string)   => new StringFieldFilterType(),
            _ when underlyingType == typeof(Guid)     => new GuidFieldFilterType(),
            _                                         => throw new NotSupportedException()
        };

**Comment by bensch8816** [comment]
I've encountered the same issue with date filtering in my projects and it can be quite a headache. It seems like there's a real need for a more flexible solution when it comes to date properties, especially when dealing with different types of date fields. I'm hopeful that a fix or a workaround will be introduced soon. Keeping an eye on this thread for updates and fingers crossed for a solution that will adress this issue.

**Comment by ElderJames** [comment]
Hello @davhev8619 @bensch8816 , please try the custom filed filter https://antblazor.com/en-US/components/table#components-table-demo-custom-field-filter

**Comment by davhev8619** [comment]
Thank you @ElderJames . I took the DateTimeFieldFilterType and removed the RemoveMilliseconds function call. It now works with MS SQL db type "date" and "datetime2". I don't know the purpose of RemoveMilliseconds .

  public class DbDateFieldFilterType : DateFieldFilterType
  {
      public override RenderFragment<TableFilterInputRenderOptions> FilterInput => FilterInputs.Instance.DateTimeInput;

      protected override Expression GetNonNullFilterExpression(TableFilterCompareOperator compareOperator,
          Expression leftExpr, Expression rightExpr)
      {
          return compareOperator switch
          {
              TableFilterCompareOperator.TheSameDateWith => Expression.Equal(
                  Expression.Property(leftExpr, nameof(DateTime.Date)),
                  Expression.Property(rightExpr, nameof(DateTime.Date))),
              _ => base.GetNonNullFilterExpression(compareOperator, leftExpr, rightExpr)
          };
      }
  }

**Comment by ElderJames** [comment]
Hello @davhev8619 @bensch8816 , we will support DateOnly/TimeOnly for table filter in [next release](https://github.com/ant-design-blazor/ant-design-blazor/pull/4034). And will release in these days.

**Reply by davhev8619** [reply]
Great! Thank you! The table-component is awesome!
